service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java

```diff
@@ -247,4 +284,205 @@ void shouldReturnNullIfC2DocumentBundleIsNotPopulated() {
+            return OrderAction.builder().type(ActionType.SEND_TO_ALL_PARTIES);
+        }
+
+        private DynamicList getDynamicList() {
+            DynamicListElement listElement = DynamicListElement.builder()
+                .label(formatLocalDateToMediumStyle(5))
+                .code(fromString("b15eb00f-e151-47f2-8e5f-374cc6fc2657"))
+                .build();
+
+            return DynamicList.builder()
+                .listItems(List.of(
+                    listElement,
+                    DynamicListElement.builder()
+                        .code(fromString("6b3ee98f-acff-4b64-bb00-cc3db02a24b2"))
+                        .label(formatLocalDateToMediumStyle(2))
+                        .build(),
+                    DynamicListElement.builder()
+                        .code(fromString("ecac3668-8fa6-4ba0-8894-2114601a3e31"))
+                        .label(formatLocalDateToMediumStyle(0))
+                        .build()))
+                .value(listElement)
+                .build();
+        }
+
+        private String formatLocalDateToMediumStyle(int i) {
+            return formatLocalDateToString(time.now().plusDays(i).toLocalDate(), FormatStyle.MEDIUM);
+        }
+    }
+
+    @Test
+    void shouldSerialiseDirectionsAsNullWhenEmptyDirections() {
+        assertThat(CaseData.builder().build()).hasFieldOrPropertyWithValue("directionsForCaseManagementOrder", null);
+    }
+
+    @Test
+    void shouldDeserialiseIndividualCmoDirectionsToDirectionsObject() throws JsonProcessingException {
+        UUID id = randomUUID();
+
+        CaseData deserialised = mapper.readValue(buildJsonDirections(id), CaseData.class);
+
+        assertThat(deserialised).isEqualToComparingFieldByField(caseDataWithPopulatedDirections(id));
+    }
+
+    @Test
+    void shouldSerialiseDirectionsObjectToIndividualDirections() throws JsonProcessingException {
+        UUID id = randomUUID();
+
+        String serialised = mapper.writeValueAsString(caseDataWithPopulatedDirections(id));
+
+        Stream.of(DirectionAssignee.values())
+            .forEach(assignee -> JSONAssert.assertEquals(getExpectedString(assignee, id), serialised, false));
+    }
+
+    private String buildJsonDirections(UUID id) throws JsonProcessingException {
+        List<Element<Direction>> directions = List.of(element(id, Direction.builder().directionType("title").build()));
+        String directionString = mapper.writeValueAsString(directions);
+
+        StringBuilder builder = new StringBuilder().append("{");
+
+        DirectionAssignee[] values = DirectionAssignee.values();
+        for (int i = 0; i < values.length; i++) {
+            DirectionAssignee assignee = values[i];
+            builder.append(getJsonDirectionForParty(assignee.toCaseManagementOrderDirectionField(), directionString));
+
+            if (i != values.length - 1) {
+                builder.append(",");
+            }
+        }
+
+        return builder.append("}").toString();
```

Comment:
Couldn't we use some serialization here, or you are afraid that it changes something? Seems not nice to handcraft json by hand.

Reply:
can use object mapper to convert a list of directions to a string instead, I guess we don't really care how the json is made though?

Reply:
Yeah we don't, just ugly if we do it by hand, it seems bad practise but find in the test code.